### ISSUE TITLE
dataDir and dataLogDir Swapped

### ISSUE THREAD

**Issue by zanes2016** [31z4/zookeeper-docker#32]
Is anyone else seeing that the dataDir and dataLogDir directories are swapped?

4-letter-word conf output:
```
$ echo conf | nc localhost 2181
clientPort=2181
dataDir=/datalog/version-2
dataLogDir=/data/version-2
...
```
Configuration file (in docker container):
```
# cat /conf/zoo.cfg 
clientPort=2181
dataDir=/data
dataLogDir=/datalog
```

An indeed, the logs are going to dataDir directory
```
# ls /data/version-2/
log.100000001
# ls /datalog/version-2/
acceptedEpoch       currentEpoch        snapshot.100000000
```

Image Details:
```
$ docker images
REPOSITORY                TAG                 IMAGE ID            CREATED             SIZE
zookeeper                 3.4.11              56d414270ae3        3 weeks ago         146MB
$ docker inspect  -f "{{.Name}} {{.Config.Cmd}}" zookeeper
/zookeeper [zkServer.sh start-foreground]
```

I assume this is not expected.

**Comment by 31z4** [comment]
@zanes2016 thanks for you report. Is this reproducable with Zookeeper running on the host instead of Docker?

**Comment by zanes2016** [comment]
@31z4 I haven't tried that yet since we run everything in docker. If it helps I can try to install/configure zookeeper to run on an Ubuntu 16.04 host.

**Comment by 31z4** [comment]
Yeah, that would be really helpful!

**Comment by 31z4** [comment]
@zanes2016 I cannot reproduce this with `zookeeper:3.4.11` docker image.

```console
# echo conf | nc localhost 2181
clientPort=2181
dataDir=/data/version-2
dataLogDir=/datalog/version-2
tickTime=2000
maxClientCnxns=60
minSessionTimeout=4000
maxSessionTimeout=40000
serverId=0

# cat /conf/zoo.cfg
clientPort=2181
dataDir=/data
dataLogDir=/datalog
tickTime=2000
initLimit=5
syncLimit=2

# ls /datalog/version-2
log.1

# ls /data
myid       version-2
```

**Comment by 31z4** [comment]
Isn't reproduced with `zookeeper:3.5`.

```console
# ls /datalog/version-2/
log.100000001

# ls /data
myid       version-2
```

Also, it seems that you were running `3.5` instead of `3.4` because of the log file name.

**Comment by 31z4** [comment]
`3.3` is also fine.

**Comment by zanes2016** [comment]
I've noticed that I only see this when run in HA/cluster mode (not single node). I'm also deploying this with Ansible, but that would be surprising if the docker_container Ansible module was causing this issue. And I am certain that I've been specifying zookeeper 3.4.

**Comment by 31z4** [comment]
Would be great to know more about how exactly do you run it. If you use `docker-compose` or `swarm mode`, please show your `stack.yml`.

**Comment by josdotso** [comment]
https://issues.apache.org/jira/browse/ZOOKEEPER-2960

**Comment by 31z4** [comment]
Fixed in `3.4.12`.
